src/reanimated2/animation/transformationMatrix/matrixUtils.tsx

```diff
@@ -0,0 +1,433 @@
+type FixedLengthArray<
+  T,
+  L extends number,
+  PassedObject = [T, ...Array<T>]
+> = PassedObject & {
+  readonly length: L;
+  [I: number]: T;
+};
+
+export type AffiniteMatrix = FixedLengthArray<FixedLengthArray<number, 4>, 4>;
+
+export type AffiniteMatrixFlat = FixedLengthArray<number, 16>;
+
+export type TransformMatrixDecomposition = Record<
+  | 'translationMatrix'
+  | 'scaleMatrix'
+  | 'rotationMatrix'
+  | 'skewMatrix'
+  | 'scaleMatrix',
+  AffiniteMatrix
+>;
```

Comment:
Why not just ... ?
```ts
export type TransformMatrixDecomposition = {
  translationMatrix: AffiniteMatrix,
  scaleMatrix: AffiniteMatrix,
  rotationMatrix: AffiniteMatrix,
  skewMatrix: AffiniteMatrix,
  scaleMatrix: AffiniteMatrix,
};
```
This way we can enforce that the object actually contains all these fields.

Reply:
It is enforced with `Record` as well, all the parameters are required. IMHO Record is more natural here since all the fields have the very same type.

Reply:
OK, but let's remove second `scaleMatrix` because currently we have two of those.